sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/models/PhoneNumberErrorConverter.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.phonenumbers.models;
+
+import com.azure.communication.phonenumbers.implementation.models.CommunicationError;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A converter between {@link CommunicationError} and
+ * {@link PhoneNumberError}.
+ */
+public final class PhoneNumberErrorConverter {
```

Comment:
I'm not sure if throwing a `PhoneNumber` exception over a cross-service `Communication` exception provides any additional value. Both are custom error types.

Reply:
Ideally, we should tell autorest: "If you see reference to common swagger, go reference common package instead of generating your own CommunicationErrorResponseException".  Unfortunately, that's not something autorest currently support. Currently, Identity and phone number each generates its own CommunicationErrorResponseException in implementation.models. This will lead to name clash in Contoso server. Not an ideal situation but unless we modify autorest to reference common, it is hard to work around that.